### ISSUE TITLE
Local snapshots reliably fail to provide data for the same graphs, not timeout dependent

### ISSUE THREAD

**Issue by zerodeux** [grafana/grafana#43628]
**What happened**: when using the "Local snapshot" feature on a dashboard with 11 graphs, I always end up with a snapshot showing 11 graphs but with no data in the 4 latest ones.

**What you expected to happen**: I'd expect to have data in 100% of the graphs.

**How to reproduce it (as minimally and precisely as possible)**: have a +10 graph dashboard, mine is 11 organized in 5 collapsable rows, trigger the local snapshot thru the UI, use whatever timeout from 4 to 120s and always obtain a working snapshot, but with no data points in the last 4 graphs.
I would gladly help to debug this since I can reproduce the problem, but I don't know how. I tried log.level=trace to no avail (only got a few unrelated lines while snapshoting).

**Anything else we need to know?**:
- it does not look timeout related : 1/ rendering of the dashboard in my browser takes ~2s, 2/ the resulting snapshot with missing data points is very reliably reproducible with the default timeout snapshot of 4s, 3/ CPU and IO usage on the server stays flat while the snapshotting is triggered.
- it does not look related to the colllapsed row bug (#26483), my dashboard does not have collapsed rows
- as a side effect/bug, the UI only reports the snapshot as completed when the programmed timeout has expired; If I ask for a 200s timeout, the snapshot is considered done 200s later and I then get the succesfull dialog with the proper link. If I ask for 4s, success in 4s. I do know the snapshot was actually ready in ~2s by peaking at the length(dashboard_snapshot.dashboard_encrypted) column while doing it (tuple appeared, size did not change with time).

**Environment**:
- Grafana version: 8.3.3
- Data source type & version: InfluxDB 1.8.10
- OS Grafana is installed on: Debian 10
- User OS & Browser: Debian 10, Chromium 90
- Grafana plugins: none
- Others:

**Comment by zerodeux** [comment]
OK this is extremely weird, but the main bug kind of solved itself.

I switched from Chromium to Firefox to check if the client side had some effect on the snapshotting feature. I logged in with the same account in Firefox, grabbed the same dashboard at the same point in time (URL copy/pasted), did the snapshoting like explained in the "bug reproduction howto" above, and it worked.

This it's client-side and browser related. But wait... Now going back to Chromium, I *can't* reproduce the bug at all. Something happened on client side while logging on a new browser, and then propagated back to the former browser. Which suggests to me something fishy was in my session and was somewhat "cleaned up" by switching browsers (?).

I still have the need-to-wait-for-timeout for the UI to report that the snapshot is done though.
